api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -104,6 +104,16 @@ type OpenStackDataPlaneNodeSetStatus struct {
 
 	// AllIPs
 	AllIPs map[string]map[infranetworkv1.NetNameStr]string `json:"AllIPs,omitempty" optional:"true"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	// ConfigChanged - Informs us if there are any differences between the deployed spec vs what the most recent version is.
+	// We can use this to inform our decisions about when to re-execute Ansible tasks.
+	ConfigChanged bool `json:"configChanged,omitempty" optional:"true"`
```

Comment:
Why do we need this flag in Status as this is only relevant to a "particular recnocile"? We can find out if it has changed just by comparing what's in the status already and what's generated,

Reply:
It's just a clear indicator to the user that there is a variance between the current spec sections for nodes and nodeTemplate. So the use case would be that a user deploys the nodeSet, later they change some random ansible variable. This would require the user to create a new `OpenStackDataPlaneDeployment` for those changes to actually be applied to the nodes. Because this isn't really intuitive for users familiar with Kubernetes, the intention behind this boolean flag is to provide users with a clear indication that they need to take additional actions for the changes to be applied.

Reply:
>  later they change some random ansible variable.

Any changes to ansible variables and for that matter anything in NodeSet spec would require new deploy with a OpenStackDataPlaneDeployment resource. I don't think we should have a flag for that. Also, that flag would be changed from true/false when for any reason NodeSet is reconciled (again) other than changes to those sections.

Reply:
The hash I'm creating is only created from `/spec/nodes` and `/spec/nodeTemplate`. So changes to anything else aren't included in this hash. This should only be set to true if the user changes something specific to the configuration of the dataplane nodes themselves. It shouldn't be changed for each and every reconcile, because the contents of `/spec/nodes` and `/spec/nodeTemplate` won't be changed each reconcile. 

If they try to change the NTP server for example. This would then change to `configChanged: true` until the user creates a new `OpenStackDataplaneDeployment` to actually roll out the changes. Once those changes have been applied, then the `configHash` will be written to the status and this `configChanged` bool will be set back to `false`.

Reply:
>  Once those changes have been applied, then the configHash will be written to the status

Oh! you're changing the `configHash` in `Status` only if `isDeploymentReady: true` (I missed that and forgot that we're now checking all existing deployments, which is not in your PR, it needs rebase). Then it's the deployed config hash (not the current one based on the spec changes), It would be be better to rename it as `DeployedConfigHash`. Also, why're we generating the configHash in every reconcile? Those lines from L196 should be moved there too.

Reply:
If we keep the flag to show that config has changed, you probably have to leave code where you calculate the `ConfigHash`, but this has to clear to users that though the flag is true and ConfigHash in status is the old one without deployment. So yeah probably renaming it to `DeployedConfigHash` would make more sense.

Reply:
Also I missed `isDeploymentReady` would be true without a new deployment (old deployments already deployed). So this probably won't work at all.

Reply:
I see what you mean, yeah. 

Maybe a better option would be to copy the configHash onto the deployment. Then if the associated deployment and the nodeSet have a variance in the configHash we could provide some indication for the user that the current config on the NodeSet has not been applied.

Reply:
Why not just ConfigHash and DeployedConfigHash?
ConfigHash is the hash of the current spec.
DeployedConfigHash is the hash at the last time a deployment was done with the NodeSet services.

If they're equal, that's the signal that no new deployment is needed, and a deployment is needed if they are not equal.